src/python/pants/backend/codegen/thrift/apache/python/subsystem.py

```diff
@@ -20,21 +20,16 @@ class ThriftPythonSubsystem(Subsystem):
             "See `thrift -help` for supported values."
         ),
     )
-    _runtime_dependencies = TargetListOption(
-        "--runtime-dependencies",
+    infer_runtime_dependency = BoolOption(
+        "--infer-runtime-dependency",
+        default=True,
         help=(
-            "A list of addresses to `python_requirement` targets for the runtime "
-            "dependencies needed for generated Python code to work. For example, "
-            "`['3rdparty/python:thrift']`. These dependencies will "
-            "be automatically added to every `thrift_source` target. At the very least, "
-            "this option must be set to a `python_requirement` for the "
-            "`thrift` runtime library."
+            "If True, will add a dependency on a `python_requirement` target exposing the `thrift` "
+            "module (usually from the `thrift` requirement).\n\n"
+            "Unless this option is disabled, will error if no relevant target is found or if >2 "
```

Comment:
Same style comments as mentioned earlier: Spell out `>2` and `Pants will error ...`